### DISCUSSION TITLE
Deploy repository error

### DISCUSSION THREAD

**Discussion by StacD** [community/community#65217] [Repositories] [answered]
### Select Topic Area

Bug

### Body

I hope someone can help I have deployed a repository but when I click it a new tab opens but shows an error message
 'error 404 there isn't a git hub pages site here'
When I go back to the deloy page the option to deploy has gone ie the save box is greyed out as tho it has indeed already been done. Any help would be greatly appreciated.

**Comment by arafajs** [comment, marked as answer]
1. Make sure that your repository is set to public. GitHub Pages sites can only be hosted from public repositories.
2. Check that you have an index.html, index.md, or README.md file in the root directory of your repository. GitHub Pages will look for one of these files to serve by default.
3. Make sure that GitHub Pages is enabled for your repository. You can do this by going to the settings page for your repository and checking the box next to "GitHub Pages".
4. Clear your browser's cache. This may help if the error is caused by a cached file.
5. Try deploying your repository again. If you have already deployed your repository, you can try deploying it again by clicking the "Deploy" button on the settings page.
6. The error message "There isn't a GitHub Pages site here" can also be caused by a typo in the repository name. Make sure that your repository name is spelled correctly.

**Reply by StacD** [reply]
Thank you so much I will have a look at these suggestions!

**Comment by GhostOf0days** [comment]
Github Pages is limited and cannot deploy dynamic websites. Please try Netlify, Heroku, Railway, or Vercel. For Netlify, make sure you have a _redirects file with '/* /index.html 200' in it. Replace index.html with the main HTML file name you have if it's a different name.
